### PR TITLE
Extension plugin support

### PR DESCRIPTION
**Pull request by th-we**
@annplaksin and I worked on how to export custom symbols and text styles as sensible MEI, without having to fork Sibmei. For this, we reworked the entire text and symbol export. This brings quite some substantial changes to the code, and the following benefits:

* Better conversion of formatting for all texts
* text substitution variables work everywhere (see #152)
* large `switch`es were replaced by declarative definitions of symbols
* fermatas were improved while porting them to the declarative approach (#89)

We wrote [documentation](https://github.com/notengrafik/sibmei/blob/f7fc603d502d02d6a4f9b1e16e0f038e69b24626/Extensions.md) and an [example](https://github.com/notengrafik/sibmei/blob/f7fc603d502d02d6a4f9b1e16e0f038e69b24626/lib/sibmei4_extension_test.plg).

### CODE REVIEW

File: src/ExportConverters.mss [hunk 1]
```diff
@@ -592,15 +592,11 @@ function ConvertSibeliusStructure (score) {
     {
         for each Bar b in s
         {
-            if (bar_to_staff.PropertyExists(b.BarNumber))
```
Comment (ahankinson) [inline src/ExportConverters.mss:595]: Are these changes part of the plugin support?

Comment (th-we) [reply src/ExportConverters.mss:595]: There are a few changes not strictly related to text, symbols and extensions that I made while working with the code.  That's the case here and in both the Run.mss files.  Some things are only for clarity and cosmetics (like here code that does the same thing, but a bit clearer; or inconsistent indentation in one Run.mss).  Some things are for easier development and use, like better file name defaults when exporting or making it possible to automatically calling the mocha tests after exporting the test set.

**Comment by annplaksin** [comment]
Are there any objections about this PR? If not, I would like to merge it maybe next week or so...

**Comment by ahankinson** [comment]
I can merge it

**Comment by annplaksin** [comment]
Thanks a lot!

**Comment by ahankinson** [comment]
THank you (and @th-we ) for the work!

**Pull request merged**